dlib/image_transforms/interpolation.h

```diff
@@ -229,94 +229,130 @@ namespace dlib
     class interpolate_bilinear
     {
     public:
-
         template <typename T, typename image_view_type, typename pixel_type>
-        typename disable_if<is_rgb_image<image_view_type>,bool>::type operator() (
-            const image_view_type& img,
-            const dlib::vector<T,2>& p,
-            pixel_type& result
-        ) const
+        bool operator() (
+            const image_view_type &img,
```

Comment:
Please use the same style as in the other parts of the code.  So & attaches to the type.